src/main/java/de/unistuttgart/iste/gits/media_service/controller/MediaController.java

```
@@ -25,63 +37,55 @@ public class MediaController {
     private final MediaUserProgressDataService mediaUserProgressDataService;
 
     @QueryMapping
-    public List<MediaRecord> mediaRecords(final DataFetchingEnvironment env) {
+    public List<MediaRecord> mediaRecords(final DataFetchingEnvironment env, @ContextValue final LoggedInUser currentUser) {
+        validateUserHasGlobalPermission(currentUser, Set.of(LoggedInUser.RealmRole.SUPER_USER));
         return mediaService.getAllMediaRecords(
                 uploadUrlInSelectionSet(env),
                 downloadUrlInSelectionSet(env)
         );
     }
 
     @QueryMapping
-    public List<MediaRecord> mediaRecordsByIds(@Argument final List<UUID> ids, final DataFetchingEnvironment env) {
-        return mediaService.getMediaRecordsByIds(
-                ids,
-                uploadUrlInSelectionSet(env),
-                downloadUrlInSelectionSet(env)
-        );
+    public List<MediaRecord> mediaRecordsByIds(@Argument final List<UUID> ids, final DataFetchingEnvironment env, @ContextValue final LoggedInUser currentUser) {
+        final List<MediaRecord> mediaRecords = mediaService.getMediaRecordsByIds(ids, uploadUrlInSelectionSet(env), downloadUrlInSelectionSet(env));
+
+        return checkAccessForMediaRecords(currentUser, mediaRecords, UserRoleInCourse.STUDENT);
```

should throw an error, not return null